geometric_kernels/spaces/hypersphere.py

```diff
@@ -0,0 +1,188 @@
+        r"""
+        Evaluates the spherical harmonics at `X`, which are Euclidian coordinates.
+        In other words, the points are parameterized by their extrinsic
+        (self.dim+1)-coordinates.
+
+        :param X: TensorType, [N, self.dim+1]
+            N points with unit norm in Euclidian coordinate system (extrinsic).
+
+        :return: [N, M], where M = self.num_eigenfunctions
+        """
+        return self._spherical_harmonics(X)
+
+    def _addition_theorem(self, X: B.Numeric, X2: B.Numeric, **parameters) -> B.Numeric:
+        r"""
+        Returns the result of applying the additional theorem when
+        summing over all the eigenfunctions within a level, for each level
+
+        Concretely, in the case for inputs on the hypersphere, summing over all the
+        spherical harmonics within a level is equal to evaluating the Gegenbauer polynomial.
+
+        :param X: [N, dim+1]
+        :param X2: [N2, dim+1]
+        :param parameters: unused.
+        :return: Evaluate the sum of eigenfunctions on each level. Returns
+            a value for each level [N, N2, L]
+        """
+        values = [
+            level.addition(X, X2)[..., None]  # [N1, N2, 1]
+            for level in self._spherical_harmonics.harmonic_levels
+        ]
+        return B.concat(*values, axis=2)  # [N, N2, L]
+
+    def _addition_theorem_diag(self, X: B.Numeric, **parameters) -> B.Numeric:
+        """
+        Returns the sum of eigenfunctions on a level for which we have a simplified expression
+
+        :param X: [N, 1]
+        :param parameters: unused.
+        :return: Evaluate the sum of eigenfunctions on each level. Returns
+            a value for each level [N, L]
+        """
+        values = [
+            level.addition_at_1(X)  # [N, 1]
+            for level in self._spherical_harmonics.harmonic_levels
+        ]
+        return B.concat(*values, axis=1)  # [N, L]
+
+    @property
+    def num_eigenfunctions(self) -> int:
+        """Number of eigenfunctions, M"""
+        return self._num_eigenfunctions
+
+    @property
+    def num_levels(self) -> int:
+        """
+        Number of levels, L
+
+        For each level except the first where there is just one, there are two
+        eigenfunctions.
+        """
+        return self._num_levels
+
+    @property
+    def num_eigenfunctions_per_level(self) -> B.Numeric:
+        """Number of eigenfunctions per level, [N_l]_{l=0}^{L-1}"""
+        return [num_harmonics(self.dim + 1, level) for level in range(self.num_levels)]
+
+
+class Hypersphere(DiscreteSpectrumSpace, gs.geometry.hypersphere.Hypersphere):
+    r"""
+    The d-dimensional hypersphere embedded in the (d+1)-dimensional Euclidean space.
+    """
+
+    def __init__(self, dim: int):
+        r"""
+        :param dim: Dimension of the hypersphere :math:`S^d`.
+        """
+        super().__init__(dim=dim)
+        self.dim = dim
+
+    def is_tangent(
```

Comment:
Why do we need `is_tangent` actually?

Reply:
Indeed seems odd, the geomstats Hypersphere has a `to_tangent` method but no way to check if it is tangent, and this method is just a stub?

Reply:
We do not have `is_tangent` for the hyperbolic space, although we do for the circle. As it returns a `NotImplementedError `in both case, do we really want to keep this?